### PR TITLE
Feature/Address recovery from signature

### PR DESCRIPTION
**Pull request by MohitKumarAgniotri**
Enabled API 
`int address_from_signature(const uint8_t *data, size_t data_len, const signature_t *signature, address_t *addr_out)`

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -95,3 +104,4 @@ add_test(test_helpers test_helpers)
 add_test(test_crypto test_helpers)
 add_test(test_tx test_tx)
 add_test(test_web3 test_tx)
```
Comment (pcppcp) [inline tests/CMakeLists.txt]: aahh, this is a bug, it should be `add_test(test_web3 test_tx)` of course

File: tests/test_address.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+/* system includes C++ */
+#include <gtest/gtest.h>
+
+/* local includes */
+#include <helpers/hextobin.h>
+#include <eth/address.h>
+#include <eth/sign.h>
+
+TEST(TEST_ADDRESS, TEST_ADDRESS_RECOVERY)
```
Comment (pcppcp) [inline tests/test_address.cpp]: I think this test would also fit in the `test_crypto` file, what do you think?

File: tests/CMakeLists.txt (hunk 1 above)
Comment (MohitKumarAgniotri) [reply tests/CMakeLists.txt]: yes, and we have #121 for the same.

File: tests/test_address.cpp (hunk 2 above)
Comment (MohitKumarAgniotri) [reply tests/test_address.cpp]: I agree we can add it to the same file and remove the new file altogether.

**Comment by MohitKumarAgniotri** [comment]
@pcppcp I think the API `address_from_signature` belongs better in the address.c as it uses the information to retrieve address unlike all the function in the sign.c who use to address to generate sign.

**Pull request merged**